agentcfg/fetch.go

```diff
@@ -66,8 +76,59 @@ func NewFetcher(client kibana.Client, cacheExpiration time.Duration) *Fetcher {
 	}
 }
 
+// ValidationError encapsulates a validation error from the Fetcher.
+// ValidationError implements the error interface.
+type ValidationError struct {
+	keyword, body string
+	err           error
+}
+
+// Keyword returns the keyword for the ValidationError.
+func (v *ValidationError) Keyword() string { return v.keyword }
+
+// Body returns the body for the ValidationError.
+func (v *ValidationError) Body() string { return v.body }
+
+// Error() implements the error interface.
+func (v *ValidationError) Error() string { return v.err.Error() }
+
+// Validate validates the currently configured Fetcher.
+func (f *Fetcher) validate(ctx context.Context) *ValidationError {
```

Comment:
Doesn't have to be in this PR, but we should rename Fetcher to KibanaFetcher or something similar.

Reply:
No time like the present